assets/scripts/aircraft.js

```diff
@@ -1162,6 +1199,54 @@ var Aircraft=Fiber.extend(function() {
       };
       return ['ok', readback];
     },
+    runClearedAsFiled: function() {
+      this.runSID(this.destination);
+      var apt = airport_get();
+      var init_alt_instr_L = "", init_alt_instr_S = "";
+
+      this.fms.setAll({altitude:apt.initial_alt});
+      var init_alt_instr_L = ". Climb and maintain " + apt.initial_alt + 
+        ", expect " + this.fms.fp.altitude + " 10 minutes after departure";
+      var init_alt_instr_S = ". Climb and maintain " + radio_altitude(apt.initial_alt) +
+        ", expect " + radio_altitude(this.fms.fp.altitude) + "," + radio_spellOut(" 10 ") +
+        "minutes after departure";
+
+      var readback = {
+        log: "cleared to destiantion via the " + airport_get().sids[this.destination].icao + 
+          " departure, then as filed" + init_alt_instr_L,
+        say: "cleared to destination via the " + airport_get().sids[this.destination].name +
+          " departure, then as filed" + init_alt_instr_S};
+      return['ok', readback];
+    },
+    runClimbViaSID: function() {
+      var wp = this.fms.waypoints;  // copy current waypoints
+      var cruise = this.fms.fp.altitude;
+
+      for(var i=0; i<wp.length; i++) {
+        var a = wp[i].fixRestrictions.alt;
+        if(a) {
+          if(a.indexOf("+") != -1) {  // at-or-above altitude restriction
+            var minAlt = parseInt(a.replace("+",""));
+            var alt = Math.min(cruise, airport_get().ctr_ceiling);
+          }
+          else if(a.indexOf("-") != -1) {
+            var maxAlt = parseInt(a.replace("-",""));
+            var alt = Math.min(alt, cruise) // climb as high as restrictions permit
+          }
+        }
+        else var alt = Math.min(cruise, airport_get().ctr_ceiling);
+
+        wp[i].altitude = alt; // add altitudes to wp
+      }
```

Comment:
Stylistically, this sort of manipulation of FMS structures means the code should probably be part of FMS instead of Aircraft.